homeassistant/components/recorder/migration.py

```diff
@@ -363,6 +363,7 @@ def _apply_update(engine, new_version, old_version):
         if engine.dialect.name == "mysql":
             _modify_columns(engine, "events", ["event_data LONGTEXT"])
             _modify_columns(engine, "states", ["attributes LONGTEXT"])
+        _modify_columns(engine, "events", ["event_type VARCHAR(64)"])
```

Comment:
```suggestion
        if instance.engine.dialect.name == "sqlite"
            # sqlite cannot alter columns
            _modify_columns(engine, "events", ["event_type VARCHAR(64)"])
```

Reply:
This fails on postgresql
```
2021-03-11 08:33:13 WARNING (Recorder) [homeassistant.components.recorder.migration] Modifying columns event_type in table events. Note: this can take several minutes on large databases and sl
2021-03-11 08:33:13 ERROR (Recorder) [homeassistant.components.recorder.util] Error executing query: (psycopg2.errors.SyntaxError) syntax error at or near "MODIFY"
LINE 1: ALTER TABLE events MODIFY event_type VARCHAR(64)
                           ^

[SQL: ALTER TABLE events MODIFY event_type VARCHAR(64)]
(Background on this error at: http://sqlalche.me/e/13/f405)
2021-03-11 08:33:13 ERROR (Recorder) [homeassistant.components.recorder] Error during connection setup to postgresql://postgres:homeassistant@77b2833f-timescaledb/homeassistant: (psycopg2.erro
LINE 1: ALTER TABLE events MODIFY event_type VARCHAR(64)
                           ^

[SQL: ALTER TABLE events MODIFY event_type VARCHAR(64)]
(Background on this error at: http://sqlalche.me/e/13/f405) (retrying in 3 seconds)
```

Reply:
https://dbfiddle.uk/?rdbms=postgres_9.5&fiddle=40f8bd4ca9c9d4cb599183668478b085

Reply:
https://dbfiddle.uk/?rdbms=postgres_9.5&fiddle=b694eb2bec4e4ff56595acc83301aea3

Reply:
I think it makes sense to only do it for mysql.  Anyone else who wants the larger size can recreate the database or do it manually.